File: frontend/src/models/MSSResult.ts
export type MSSDirection = "BULLISH" | "BEARISH" | "NONE";

export interface MSSResult {
  valid: boolean;
  direction: MSSDirection;
  brokenStructureLevel: number | null;
  liquiditySweep: boolean;
  displacement: boolean;
  fvgCreated: boolean;
  biasAligned: boolean;
  passedRules: string[];
  failedRules: string[];
  reasons: string[];
  warnings: string[];
}

File: backend/app/engines/mss/MSSResult.py
from __future__ import annotations

from typing import Literal
from typing import TypedDict


class MSSResult(TypedDict):
    valid: bool
    direction: Literal["BULLISH", "BEARISH", "NONE"]
    brokenStructureLevel: float | None
    liquiditySweep: bool
    displacement: bool
    fvgCreated: bool
    biasAligned: bool
    passedRules: list[str]
    failedRules: list[str]
    reasons: list[str]
    warnings: list[str]
